pkg/windows/windows.go

```diff
@@ -391,6 +404,26 @@ func (vm *windows) FileExists(path, checksum string) (bool, error) {
 	return false, nil
 }
 
+func (vm *windows) ReplaceDir(files map[string][]byte, remoteDir string) error {
+	vm.log.V(1).Info("overwriting", "remote destination dir", remoteDir, "number of files", len(files))
+
+	if out, err := vm.Run(rmDirCmd(remoteDir), true); err != nil {
+		return fmt.Errorf("unable to remove directory %s, out: %s, err: %s", remoteDir, out, err)
+	}
+	if out, err := vm.Run(mkdirCmd(remoteDir), false); err != nil {
+		return fmt.Errorf("unable to create remote directory %s, out: %s: %w", remoteDir, out, err)
+	}
+
+	sftpClient, err := vm.interact.createSFTPClient()
+	if err != nil {
+		return fmt.Errorf("Failed to create SFTP client: %w", err)
+	}
+	if err = vm.interact.transferFiles(sftpClient, files, remoteDir); err != nil {
+		return err
+	}
+	return sftpClient.Close()
```

Comment:
shouldn't the client be closed even if we return error in between? This also applies to  a few other places in this commit.